Keas.Mvc/Services/ReportService.cs

```diff
@@ -448,5 +454,136 @@ public async Task<IList<PeopleLeavingWithAssetsModel>> PeopleLeavingWithAssets(G
+            }
+
+            var expiringEquipment = await _context.Equipment.Where(a => (showType == "All" || showType == "Equipment") &&
+                a.Team.Groups.Any(g => g.GroupId == group.Id) &&
+                a.Assignment.ExpiresAt <= expiresBefore)
+                .Include(a => a.Team).Include(e => e.Assignment).ThenInclude(a => a.Person).Select(ExpiringEquipmentProjection()).ToArrayAsync();
+
+            if (expiringEquipment.Any())
+            {
+                model.ExpiringItems.AddRange(expiringEquipment);
+            }
+
+            var expiringWorkstations = await _context.Workstations.Where(a => (showType == "All" || showType == "Workstation") &&
+                a.Team.Groups.Any(g => g.GroupId == group.Id) && 
+                a.Assignment.ExpiresAt <= expiresBefore)
+                .Include(a => a.Team).Include(w => w.Assignment).ThenInclude(a => a.Person).Select(ExpiringWorkstationProjection()).ToArrayAsync();
+
+            if (expiringWorkstations.Any())
+            {
+                model.ExpiringItems.AddRange(expiringWorkstations);
+            }
+
+            return model;
+        }
+
+        public Expression<Func<AccessAssignment, ExpiringItemReportModel>> ExpiringAccessProjection()
+        {
+            return a => new ExpiringItemReportModel
+            {
+                Type = "Access",
+                ItemName = a.Access.Name,
+                PersonName = a.Person.Name,
+                ExpiresAt = a.ExpiresAt.ToPacificTime(),
+                TeamSlug = a.Access.Team.Slug,
+                DetailsLink = $"/{a.Access.Team.Slug}/access/details/{a.AccessId}", //This is really only needed for the non group one?
+            };
+        }
+
+        public Expression<Func<KeySerial, ExpiringItemReportModel>> ExpiringKeysProjection()
+        {
+            return a => new ExpiringItemReportModel
+            {
+                Type = "Key",
+                ItemName = $"{a.Key.Name} {a.Key.Code}-{a.Number}",
+                PersonName = a.KeySerialAssignment.Person.Name,
+                ExpiresAt = a.KeySerialAssignment.ExpiresAt.ToPacificTime(),
+                TeamSlug = a.Team.Slug,
+                DetailsLink = $"/{a.Team.Slug}/keys/details/{a.KeyId}/keyserials/details/{a.KeySerialAssignment.KeySerialId}", //This is really only needed for the non group one?
+            };
+        }
+        public Expression<Func<Equipment, ExpiringItemReportModel>> ExpiringEquipmentProjection()
+        {
+            return a => new ExpiringItemReportModel
+            {
+                Type = "Equipment",
+                ItemName = a.Name,
+                PersonName = a.Assignment.Person.Name,
+                ExpiresAt = a.Assignment.ExpiresAt.ToPacificTime(),
+                TeamSlug = a.Team.Slug,
+                DetailsLink = $"/{a.Team.Slug}/equipment/details/{a.Id}", //This is really only needed for the non group one?
+            };
+        }

```

Comment:
I do like this model though, declaring an expression for a projection and then using it where needed instead of a one-time inliner